### PR TITLE
Importing coverage-ensure in to master (Cake3) branch.

### PR DESCRIPTION
**Pull request by beporter**
Ref: #43

### CODE REVIEW

**Comment by beporter** [comment]
TODO: Need to modify the path detection for the phpunit.xml file. Also, since we're now using phpunit.xml.dist, there is a chance that there may be a phpunit.xml that overrides it on a per-developer basis. We'll need to handle that case too.

**Comment by beporter** [comment]
Can someone (`*cough*` @ricog) try copying this script into a Cake 3 project to make sure I got the `phpunit.xml` file detection changes right? No rush.

**Comment by ricog** [comment]
I confirmed this works when specifying the clover.xml. I'll confirm the file detection when I get a chance.

**Comment by beporter** [comment]
Just to clarify, it should work in a couple of cases:

* Command line-provided clover.xml path: Takes precedence over everything and is always used when present.
* No phpunit.xml or phpunit.xml.dist file: No guessing done, errors when no clover.xml provided on command line.
* Either phpunit.xml or phpunit.xml.dist present: Obtains clover.xml path from the `<logging>` section, with preference to the phpunit.xml file.

In other words, it should "work" with a provided path (regardless of phpunit.xml[.dist] files), either single file, or both files.
